columnflow/tasks/framework/mixins.py

```diff
@@ -2157,8 +1936,173 @@ def processes_repr(self):
         return f"{len(self.processes)}_{law.util.create_hash(self.processes)}"
 
 
-class ShiftSourcesMixin(ConfigTask):
+class MultiConfigDatasetsProcessesMixin(AnalysisTask):
+    # NOTE: changed from CSV to MultiCSVParameter, might break things
+    # where self.dataset or self.dataset_inst is used
+    datasets = law.MultiCSVParameter(
+        default=(),
+        description="comma-separated dataset names or patters to select; can also be the key of a "
+        "mapping defined in the 'dataset_groups' auxiliary data of the config; when empty, uses "
+        "all datasets registered in the config that contain any of the selected --processes; empty "
+        "default",
+        brace_expand=True,
+        parse_empty=True,
+    )
 
+    # TODO: this should be per config
+    processes = law.MultiCSVParameter(
+        default=(),
+        description="comma-separated process names or patterns for filtering processes; can also "
+        "be the key of a mapping defined in the 'process_groups' auxiliary data of the config; "
+        "uses all processes of the config when empty; empty default",
+        brace_expand=True,
+        parse_empty=True,
+    )
+
+    allow_empty_datasets = False
+    allow_empty_processes = False
+
+    @classmethod
+    def resolve_param_values(cls, params):
+        params = super().resolve_param_values(params)
+
+        config_insts = params.get("config_insts", ())
+
+        if not config_insts:
+            # configs not yet setup
+            return params
+
+        n_configs = len(config_insts)
+
+        # resolve processes
+        # TODO: should be resolved for all configs
+        if "processes" in params:
+            if params["processes"]:
```

Comment:
I am not so sure about this. There is a difference between `processes` being empty and having no `processes` key present at all. With the current implementation, we only raise an exception when the `processes` key is set and it is resolved to an empty tuple, but not when there is no `processes` key to begin with. I suppose this might happen if some class inherits from this mixin, but disables the `processes` parameter.